src/backend/storeManagers/legendary/library.ts

```diff
@@ -666,7 +666,11 @@ export async function runRunnerCommand(
   if (!options.env) {
     options.env = {}
   }
-  options.env.XDG_CONFIG_HOME = dirname(legendaryConfigPath)
+
+  // if not on a SNAP environment, set the XDG_CONFIG_HOME to the same location as the config file
+  if (!process.env.SNAP) {
+    options.env.XDG_CONFIG_HOME = dirname(legendaryConfigPath)
```

Comment:
does chaning XDG_CONFIG_HOME under snap cause issues? if it does does it still cause issues if a different variable is used?

Reply:
Yes, it breaks the SNAPs because then legendary was trying to write its configs to a place that it is not possible so it was giving permission issues.